server/gRPC.go

```diff
@@ -0,0 +1,152 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/url"
+
+	cliContext "github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/maticnetwork/heimdall/helper"
+	"github.com/maticnetwork/heimdall/server/proto"
+	tmLog "github.com/tendermint/tendermint/libs/log"
+	"google.golang.org/grpc"
+)
+
+const (
+	LatestSpanURL   = "/bor/latest-span"
+	SpanURL         = "/bor/span/%v"
+	EventRecordList = "/clerk/event-record/list"
+)
+
+var logger tmLog.Logger
+
+func setupGRPCServer(shutDownCtx context.Context, cdc *codec.Codec, addr string, lggr tmLog.Logger) error {
+	logger = lggr
```

Comment:
Love it! It is a great first step to have gRPC.

Could we have https://github.com/grpc-ecosystem/go-grpc-middleware with proper middlewares as next?

Reply:
Sure, Working on all the above comments,Thanks for the review.

Reply:
I have added one interceptor based on what I saw in bor code, Looking to add more like prometheus.

Reply:
Didn't found much use of any other as of now, Will review middleware part later once integrated from bor side as well.